pkg/server/in_cluster_application.go

```diff
@@ -0,0 +1,46 @@
+package server
+
+import (
+	"github.com/codeready-toolchain/registration-service/pkg/application"
+	"github.com/codeready-toolchain/registration-service/pkg/application/service"
+	servicecontext "github.com/codeready-toolchain/registration-service/pkg/application/service/context"
+	"github.com/codeready-toolchain/registration-service/pkg/application/service/factory"
+	"github.com/codeready-toolchain/registration-service/pkg/configuration"
+	"github.com/codeready-toolchain/registration-service/pkg/kubeclient"
+	"k8s.io/client-go/rest"
+)
+
+// NewInClusterApplication creates a new in-cluster application with the specified configuration and options.  This
+// application type is intended to run inside a Kubernetes cluster, where it makes use of the rest.InClusterConfig()
+// function to determine which Kubernetes configuration to use to create the REST client that interacts with the
+// Kubernetes service endpoints.
+func NewInClusterApplication(config configuration.Configuration, options ...factory.Option) (application.Application, error) {
+	app := new(InClusterApplication)
+
+	k8sConfig, err := rest.InClusterConfig()
+	if err != nil {
+		return nil, err
+	}
+
+	kubeClient, err := kubeclient.NewCRTRESTClient(k8sConfig, config.GetNamespace())
+	if err != nil {
+		return nil, err
+	}
+
+	app.serviceFactory = factory.NewServiceFactory(func() servicecontext.ServiceContext {
+		return factory.NewServiceContext(kubeClient, config)
+	}, config, options...)
```

Comment:
You're right, and I actually have a chicken and egg problem here.. I'll think of some way I can reuse the same `ServiceFactory` instance.

Reply:
This was a tricky one, I had to spend most of yesterday and today working through this problem because there was no easy solution.  I've ended up refactoring a significant amount of code as a result, and now there is only a single service factory created and reused for the service contexts.